src/Cutscenes.ts

```diff
@@ -1396,6 +1396,130 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
     }
 
     /**
+     * Cutscene for a Ice Beam attack in battle.
+     * 
+     * @param settings   Settings used for the cutscene.
+     * @param args   Settings for the routine.
+     */
+    public cutsceneBattleAttackIceBeam(settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+        const defenderName: string = args.defenderName;
+        const attackerName: string = args.attackerName;
+        const defender: IThing = this.EightBitter.BattleMover.getThing(defenderName) as IThing;
+        const attacker: IThing = this.EightBitter.BattleMover.getThing(attackerName) as IThing;
+        const dt: number = 1;
+        const size: number = this.EightBitter.unitsize;
+        const direction: number = defenderName === "opponent" ? -1 : 1;
+        const lineArray: IThing [] = [];
+        const menu: IMenu = this.EightBitter.MenuGrapher.getMenu("BattleDisplayInitial") as IMenu;
+        const icicle1: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles1");
+        const icicle2: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles2");
+        const icicle3: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles3");
+        const icicle4: IThing = this.EightBitter.ObjectMaker.make("IceBeamIcicles4");
```

Comment:
Turn these into an array. If you ever have variables with numbers in the name that's a sure sign.